tgui/packages/tgui/interfaces/GPS.js

```diff
@@ -0,0 +1,223 @@
+      <Window.Content>
+        <Flex direction="column" height="100%">
+          {emped ? (
+            <Flex.Item grow="1" basis="0">
+              <TurnedOff emp />
+            </Flex.Item>
+          ) : (
+            <Fragment>
+              <Flex.Item>
+                <Settings />
+              </Flex.Item>
+              {active ? (
+                <Fragment>
+                  <Flex.Item mt="0.5rem">
+                    <Position area={area} position={position} />
+                  </Flex.Item>
+                  {saved && (
+                    <Flex.Item mt="0.5rem">
+                      <Position title="Saved Position" position={saved} />
+                    </Flex.Item>
+                  )}
+                  <Flex.Item mt="0.5rem" grow="1" basis="0">
+                    <Signals height="100%" />
+                  </Flex.Item>
+                </Fragment>
+              ) : (
+                <TurnedOff />
+              )}
+            </Fragment>
+          )}
+        </Flex>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const TurnedOff = ({ emp }, context) => {
+  return (
+    <Section
+      mt="0.5rem"
+      width="100%"
+      height="100%"
+      stretchContents>
+      <Box
+        width="100%"
+        height="100%"
+        color="label"
+        textAlign="center">
+        <Flex height="100%">
+          <Flex.Item grow="1" align="center" color="label">
+            <Icon
+              name={emp ? "ban": "power-off"}
+              mb="0.5rem"
+              size="5"
+            /><br />
+            {emp
+              ? "ERROR: Device temporarily lost signal."
+              : "Device is disabled."}
+          </Flex.Item>
+        </Flex>
+      </Box>
+    </Section>
+  );
+};
+
+const Settings = (properties, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    active,
+    tag,
+    same_z,
+  } = data;
+  const [newTag, setNewTag] = useLocalState(context, "newTag", tag);
+  return (
+    <Section
+      title="Settings"
+      buttons={
+        <Button
+          selected={active}
+          icon={active ? "toggle-on" : "toggle-off"}
+          content={active ? "On" : "Off"}
+          onClick={() => act('toggle')}
+        />
+      }>
+      <LabeledList>
+        <LabeledList.Item label="Tag">
+          <Input
+            width="5rem"
+            value={tag}
+            onEnter={() => act('tag', { newtag: newTag })}
+            onInput={(e, value) => setNewTag(value)}
+          />
+          <Button
+            disabled={tag === newTag}
+            width="20px"
+            mb="0"
+            ml="0.25rem"
+            onClick={() => act('tag', { newtag: newTag })}>
+            <Icon name="pen" />
+          </Button>
+        </LabeledList.Item>
+        <LabeledList.Item label="Range">
+          <Button
+            selected={!same_z}
+            icon={same_z ? "compress" : "expand"}
+            content={same_z ? "Normal" : "Extended"}
```

Comment:
`Normal` and `Extended` dont seem right here. Maybe `Local Sector` and `Global`?